synapse/rest/client/read_marker.py

```diff
@@ -40,9 +40,12 @@ def __init__(self, hs: "HomeServer"):
         self.read_marker_handler = hs.get_read_marker_handler()
         self.presence_handler = hs.get_presence_handler()
 
-        self._known_receipt_types = {ReceiptTypes.READ, ReceiptTypes.FULLY_READ}
-        if hs.config.experimental.msc2285_enabled:
-            self._known_receipt_types.add(ReceiptTypes.READ_PRIVATE)
```

Comment:
I think we still only want to support the unstable version if it is enabled in the experimental config?

Reply:
I don't think I see the benefit of doing that? If we have the same handling for both is there a reason to do it? At least in Element Web we're going to delabs it both with stable and unstable...

I might be wrong though - am I missing anything?

Reply:
Also, if we actually decided to go with this - what exactly should be controlled by the config?

Reply:
Also, if do add this, it could confuse a bunch of people - why is the flag not showing up in Element I've updated my Synapse?

Reply:
> I don't think I see the benefit of doing that? If we have the same handling for both is there a reason to do it? At least in Element Web we're going to delabs it both with stable and unstable...
> 
> I might be wrong though - am I missing anything?

The benefit is that the unstable identifiers stay disabled for a Synapse deployment which did not explicitly opt into the experimental flag. I think that this makes more sense than enabling the unstable identifiers for only a short period of time.

Reply:
> Also, if do add this, it could confuse a bunch of people - why is the flag not showing up in Element I've updated my Synapse?

It would show up though since the stable identifiers would be supported?

Reply:
> > Also, if do add this, it could confuse a bunch of people - why is the flag not showing up in Element I've updated my Synapse?
> 
> It would show up though since the stable identifiers would be supported?

Not if you had an older version of Element

Reply:
> > I don't think I see the benefit of doing that? If we have the same handling for both is there a reason to do it? At least in Element Web we're going to delabs it both with stable and unstable...
> > I might be wrong though - am I missing anything?
> 
> The benefit is that the unstable identifiers stay disabled for a Synapse deployment which did not explicitly opt into the experimental flag. I think that this makes more sense than enabling the unstable identifiers for only a short period of time.

Ah, that makes sense

<hr>

So we just don't allow receiving of unstable private RRs with the feature disabled or is there anything else?

Reply:
> So we just don't allow receiving of unstable private RRs with the feature disabled or is there anything else?

I think that's pretty much all we need to do.